app/controllers/posts_controller.rb

```diff
@@ -28,6 +28,22 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find_by(id: params[:id])
+    if @post
+      @post.likes.destroy_all
+      if @post.destroy
+        flash[:notice] = 'Post successfully deleted.'
+      else
+        flash[:alert] = 'Failed to delete the post.'
+      end
+    else
+      flash[:alert] = 'Post not found.'
+    end
+
+    redirect_to user_posts_path(id: current_user.id)
+  end
+
```

Comment:
- [ ] as already suggested by the previous CR you should consider using the `load_and_authorize_resource` method provided by the CanCanCan gem in the posts_controller to load and authorize resources, this might look something like: 
```
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end

```

Reply:
@mikethreels Hi, Thanks for your feedback I we consume all our review could you please confirm this changes ok?
![Uploading Capture1.PNG…]()

Reply:
Dear Reviwer kindly consider this is out extra review